rwengine/src/data/GameTexts.hpp

```diff
@@ -3,21 +3,42 @@
 #include <string>
 #include <unordered_map>
 
+/**
+ * The game stores strings as 16-bit indexes into the font
+ * texture, which is something simllar to ASCII.
+ */
+using GameString = std::basic_string<char16_t>;
+/**
+ * Keys are small so should be subject to SSO
+ */
+using GameStringKey = std::string;
+
+namespace GameStringUtil
+{
+/**
+ * @brief fromString Converts a string to a GameString
+ *
+ * Encoding of GameStrings depends on the font, only simple ASCII chars will map well
+ */
+GameString fromString(const std::string& str);
+}
+
 class GameTexts
 {
-	std::unordered_map<std::string, std::string> _textDB;
+	using StringTable = std::unordered_map<GameStringKey, GameString>;
+	StringTable m_strings;
 public:
 
-	void addText(const std::string& id, const std::string& text) {
-		_textDB.insert({ id, text });
+	void addText(const GameStringKey& id, GameString&& text) {
```

Comment:
I'm not sure what you mean here. `id` is already just a set of single byte chars.